web/views.py

```diff
@@ -200,18 +207,100 @@ def form_valid(self, form):
         return super().form_valid(form)
 
 
-class StudiesListView(generic.ListView):
+class StudiesListView(generic.ListView, PaginatorMixin):
     """
     List all active, public studies.
     """
 
+    form_class = forms.StudyListSearchForm
     template_name = "web/studies-list.html"
     model = Study
 
+    def post(self, request, *args, **kwargs):
+        return super().get(request, *args, **kwargs)
+
     def get_queryset(self):
-        # TODO if we need to filter by study demographics vs user demographics
-        # or by if they've taken the study before this is the spot
-        return super().get_queryset().filter(state="active", public=True).order_by("?")
+        user = self.request.user
+        qs = super().get_queryset().filter(state="active", public=True)
+        page = self.request.GET.get("page", 1)
+
+        # Form will return different value depending if the user is logged in.
+        form = self.search_form()
+
+        # values from form
+        search_value = form.data.get("search", "")
+        children_value = form.data.get("children", "")
+        show_experiments_already_done_value = (
+            form.data.get("show_experiments_already_done") == "on"
+        )
+
+        if search_value:
+            qs = qs.filter(name__icontains=search_value)
+
+        # convert to list as it's no longer being treated as a queryset
+        studies = list(qs)
+
+        if children_value:
+            if user.is_anonymous:
+                # when user is anonymous, children value is an age range.
+                age_range = [int(c) for c in children_value.split(",")]
+                studies = [
+                    s for s in studies if age_range_eligibility_for_study(age_range, s)
+                ]
+            else:
+                # when user is authenticated, children value is a child pk
+                child = Child.objects.get(pk=children_value)
```

Comment:
should check somewhere that this child pk is actually associated with the logged-in account. Otherwise you could send sequential pks and determine eligibility for any arbitrary kid... probably not a security issue on its own, but in combination with any similar unintended abilities it could be (e.g. getting a list of nicknames and approximate birthdates of kids who definitely have autism since they're eligible for such-and-such study)